src/bulk/common.ts

```diff
@@ -1169,6 +1164,10 @@ export abstract class BulkOperationBase {
     return this.s.bsonOptions;
   }
 
+  get writeConcern(): WriteConcern | undefined {
```

Comment:
was this added so you can use `BulkOperationBase` as an `OperationParent`?

Reply:
Technically we could already use it as an `OperationParent` (`writeConcern` is an optional property), but this change was so `resolveOptions` can correctly pull the `writeConcern` from `BulkOperationBase`.

Happy to change the argument type in `resolveOptions` to include `BulkOperationBase` or do something else to make this more visible.

Reply:
I was actually thinking that maybe you could add a `readPreference` property which always returns `primary`, not sure if that helps at all.. Does it make sense to explicitly have things we consider operation parents `implement OperationParent`?

Reply:
I don't think it helps to add a `readPreference` property because we actually need to modify the read pref in `options`, not the parent read preference, to avoid defaulting to a transaction read pref when doing a write. Going to do what you suggested and move the read pref resolving and corresponding options modifying to `OperationBase`, and that solves the problem.

As for the `OperationParent` issue, I am not sure if we really consider `BulkOperationBase` an `OperationParent`? Similarly, it doesn't really seem like `OperationBase` is an `OperationParent`. This is why I was wondering about just adding `BulkOperationBase` as an accepted type for the parent in `resolveOptions`

Reply:
SGTM